### PR TITLE
Aliu/ci fix

### PR DESCRIPTION
**Pull request by aliuTT**

### CODE REVIEW

**Review by tt-rkim — APPROVED** [review]

**Comment by tt-rkim** [comment]
Damn I love approving stuff regarding software that I know inside-out like watcher and debug print. Very confident this will pass

**Comment by aliuTT** [comment]
@tt-dma 
This PR has your fix for DPRINT unlocking and the ethernet core watcher assert. I think pipelines are as clean as can be
https://github.com/tenstorrent-metal/tt-metal/actions/runs/7698824418

The grayskull perf issues are on main, and don't seem to be related. The other two wormhole_b0 fails are no-runners-picking-up-job type of fail. I recommend just merging this, since these changes pass locally and pass the eye check 👀

File: tt_metal/hw/inc/debug/sanitize_noc.h [hunk 1]
```diff
@@ -47,7 +47,7 @@ extern uint8_t noc_index;
 #define DEBUG_VALID_PCIE_ADDR(a, l) (((a) >= NOC_PCIE_ADDR_BASE) && ((a) + (l) <= NOC_PCIE_ADDR_END) && ((a) + (l) > (a)))
 #define DEBUG_VALID_DRAM_ADDR(a, l) (((a) >= NOC_DRAM_ADDR_BASE) && ((a) + (l) <= NOC_DRAM_ADDR_END) && ((a) + (l) > (a)))
 
-#define DEBUG_VALID_ETH_ADDR(a, l) (((a) >= MEM_ETH_BASE) && ((a) + (l) < MEM_ETH_BASE + MEM_ETH_SIZE))
+#define DEBUG_VALID_ETH_ADDR(a, l) (((a) >= MEM_ETH_BASE) && ((a) + (l) <= MEM_ETH_BASE + MEM_ETH_SIZE))
```
Comment (tt-dma) [inline tt_metal/hw/inc/debug/sanitize_noc.h:50]: Do we want to include the same `((a) + (l) > (a))` check as PCIE/DRAM as well?

**Review by tt-dma — APPROVED** [review]

File: tt_metal/hw/inc/debug/sanitize_noc.h (hunk 1 above)
Comment (aliuTT) [reply tt_metal/hw/inc/debug/sanitize_noc.h:50]: makes sense to me

**Pull request merged**